### PR TITLE
docs: refresh web design, fix several html, css and content issues

### PR DESCRIPTION
**Pull request by tbreuss**
Since there have been discussions about website re-design for a long time, but it did not really move forward, I have been working on this website cleanup branch. Mainly, because some things are a bit embarrassing.

A preview release can be visited here: https://www.mithriljs.tebe.ch/

I see this as a foundation for the next bigger web design update.

## Motivation and Context

- keep the current web design
- make the current web design more appealing
- fix obvious design/UI/UX flaws
- fix obviously wrong content
- add or refresh content where necessary

## How Has This Been Tested?

The edits have been manually tested on my local machine and the preview release (see above).

## Types of changes

- ~~Bug fix (non-breaking change which fixes an issue)~~
- ~~New feature (non-breaking change which adds functionality)~~
- ~~Breaking change (fix or feature that would cause existing functionality to change)~~
- [x] Documentation change

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation. Well, it is in fact the documentation itself ;-)
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- ~~I have added tests to cover my changes.~~
- [x] All new and existing tests passed.
- [x] I have updated `docs/changelog.md`

## Changes Details

-  link logo with homepage (better ux imho, and it is simply expected)
-  set overall line-height to 1.5 (more appealing)
-  fix all broken anchor links from menu and within content (this was annoying)
-  open all external links in a new tab (better ux imho)
-  in menu remove intendation of first level, add margin bottom to section (more space for 3rd level menu items)
-  reduce font size for code examples highlighted by prisma.js to same size as flems examples (so, font-sizes of normal preformated text, code examples and flems examples are the same size)
-  increase margin bottom for code examples to approx. 150% of p-tags (more appealing)
-  remove all "how to contribute" menu items und place it under the page as table of contents (those menu item link labels were way too long)
-  remove "code" formattings in table cells except for the description (makes it more readable)
-  for all backgrounds use a similar color tone, the one from prisma.js (so we have a simple color scheme)
-  made the hamburger a bit larger (better ux)
-  fixed z-index issue with flems example laying on top of the menu layer (obvious ux issue) 
-  fixed issue with different line heights in paragraphs and other tags when using "code" tags (keep consistent line-height)
-  in prisma.js fixed wrong background color for operators (looks to me like a bug in prisma.js)
-  use "white-space:nowrap" and "text-overflow:ellipsis" for left menu items (more appealing, better readability)
-  remove weird looking link labels when using "code" within "a" tags (this was terrible, much better now with better readability)
-  remove weird looking link labels when using "code" within "a" within "h" tags (same as above)
-  add mouse-over for "a>code"-tag (makes it clearer that it is a link)
-  adjust indentation of ordered and unordered lists (better design and imho cleaner)
-  break line of very long \<code\> examples within p-tags on small screen (broke the layout)
-  remove link to "jsonp" page (page was removed at some point)
-  on small screen hide first ruler after h1 consequently (some pages have a ruler, some not, depending wether there are menu items for the page, unnecessary inconsistency)
-  cleanup header hierarchy on certain pages (code-of-conduct, contributing, integrating-libs, releasing)
-  add a propper 404 page (nicer for the visior, with additional help maybe)
-  remove all release candidate versions from the version selector (not needed, as each version can be accessed by entering the URL in the address bar)
-  remove Mithril.js Jobs from Misc section (linked wiki page is really outdated)
-  reordered the "Misc" section (more "logical" now)
-  move `ES6+ on legacy browsers` from Resources to Misc section (to make it less prominent)
-  move `Learning Resources` from Getting Started to Misc section (to make it less prominent)
-  remove `v1 Documentation` and `v0.2 Documentation` from Misc section (we have those links already in the version selector besides the logo, and we don't want to promote older versions, of course)
-  cleanup left menu items
-  Change footer "License: MIT. © Leo Horie." to something more appropriate (better and current information with link to community)
-  add sticky footer (looks better on pages with little content)
-  add smooth scroll behavior (it's nicer) 
-  #2833
-  #2832 
-  remove "[ES6+ on legacy browsers](https://www.mithriljs.tebe.ch/es6.html#es6+-on-legacy-browsers) from menu (@modir). Instead removing we moved it to the misc section
-  slightly lighter base font color #181818 (looks lighter)
-  memorize scroll position of left column across page loads (better ux) 
-  bump prism.js up to latest 1.29.0 version
-  make two col layout with 100% height and scrollbars (in progress)
-  add "two columns 100% height layout with scrollbars" like here: https://mithriljs.tebe.ch/draft-01/
-  version selector: all we really need is latest v1.x and latest v2.x. Not sure it's worth listing minor versions (@dead-claudia)

### CODE REVIEW

File: docs/contributing.md [hunk 1]
```diff
@@ -3,35 +3,31 @@ Contribution guide for Mithril.js
...
+* [Why does Mithril.js use its own testing framework and not Mocha/Jasmine/Tape?](#why-does-mithriljs-use-its-own-testing-framework-and-not-mochajasminetape?)
+* [Why doesn't the Mithril.js codebase use ES6 via Babel or Bublé? Would a PR to upgrade be welcome?](#why-doesn't-the-mithriljs-codebase-use-es6-via-babel-or-bublé?-would-a-pr-to-upgrade-be-welcome?)
+* [Why doesn't the Mithril.js codebase use trailing semi-colons? Would a PR to add them be welcome?](#why-doesn't-the-mithriljs-codebase-use-trailing-semi-colons?-would-a-pr-to-add-them-be-welcome?)
+* [Why does the Mithril.js codebase use a mix of `instanceof` and `typeof` checks instead of `Object.prototype.toString.call`, `Array.isArray`, etc? Would a PR to refactor those checks be welcome?](#why-does-the-mithriljs-codebase-use-a-mix-of-instanceof-and-typeof-checks-instead-of-objectprototypetostringcall,-arrayisarray,-etc?-would-a-pr-to-refactor-those-checks-be-welcome?)
+* [What should I know in advance when attempting a performance related contribution?](#what-should-i-know-in-advance-when-attempting-a-performance-related-contribution?)
+* [Do you all accept donations?](#do-you-all-accept-donations?)
```
Comment (dead-claudia) [inline docs/contributing.md:19]: Could you explain the reasoning for changing the bullet point style?

**Review by dead-claudia — CHANGES_REQUESTED** [review]
See comment. Not sure the change is justified.

File: docs/contributing.md (hunk 1 above)
Comment (tbreuss) [reply docs/contributing.md:19]: When using stars for an unordered list, it has the effect, that the list is rendered as a table of contents on top of the content and not as menu items on the left hand side. But I admit that this solution is anything but obvious.

Is there a better solution? Maybe write plain markup?

Comment (tbreuss) [reply docs/contributing.md:19]: When using stars for an unordered list, it has the effect, that the list is rendered as a table of contents on top of the content and not as menu items on the left hand side. But I admit that this solution is anything but obvious.

Is there a better solution? Maybe write plain markup?

Comment (tbreuss) [reply docs/contributing.md:19]: Ok, I now see the magic behind.

1) If a page has a H1 directly followed by an UL, the items are rendered as menu items on the left hand side. The list doesn't appear on top of the content.

Example: https://mithril.js.org/index.html

2) If a page has a H1, followed by a P as an introduction or something, the UL list is rendered as table of contents.

Example: https://mithril.js.org/examples.html

Comment (tbreuss) [reply docs/contributing.md:19]: @dead-claudia I will revert this and add some introductory text.

**Review by JAForbes — APPROVED** [review]
🎉

**Comment by dead-claudia** [comment]
Closing. Please recreate this PR against https://github.com/MithrilJS/docs.

I preserved history when I created that repo, so you may be able to just `git subtree` it over and rebase/cherry-pick your way through most of it, rather than completely redoing it all from scratch.

Do be aware that your version selector should be redone, as the repo is no longer meaningfully versioned.

**Comment by dead-claudia** [comment]
@tbreuss Forgot to ping you ⬆️

**Comment by tbreuss** [comment]
Thanks @dead-claudia. I'm not sure if I can reproduce this properly. The changes were quite extensive. But I will at least try after my vacation in mid-October.

**Pull request closed without merging**